File: station-finder-backend/src/main/java/com/ganesh/finder/dto/StationMarker.java
package com.ganesh.finder.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StationMarker {
    private Long id;
    private String name;
    private Double latitude;
    private Double longitude;
    private Boolean available;
}

File: android/app/src/main/java/com/ganesh/ev/data/model/StationMarker.kt
package com.ganesh.ev.data.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

/**
 * Lightweight model for map markers — matches backend StationMarkerDTO. Only contains data needed
 * to render pins on the map.
 */
data class StationMarker(
        val id: Long,
        val name: String,
        val latitude: Double,
        val longitude: Double,
        @SerializedName("available") val available: Boolean
) : Serializable
